src/smpi/smpi_replay.cpp

```diff
@@ -382,24 +476,35 @@ static void action_test(const char *const *action){
     flag = smpi_mpi_test(&request, &status);
 
     XBT_DEBUG("MPI_Test result: %d", flag);
-    /* push back request in vector to be caught by a subsequent wait. if the test did succeed, the request is now nullptr.*/
-    get_reqq_self()->push_back(request);
 
+    //We can't call register_request, because request may be NULL.
+    //FIXME? Is this really needed? Does xbt_dict_t allow you to inser a null
```

Comment:
yes, putting null values is OK. That's a bit difficult to retrieve it (and make the difference between not finding and finding the null value), but that's possible.